plugins/localStyles.js

```diff
@@ -0,0 +1,359 @@
+    }
+
+    // Apply the new styles to their matched elements
+    // Global variables used: matchedElements
+    function reapplyElementStyles() {
+      var matchedElementsUniq  = uniq(matchedElements);
+      matchedElementsUniq.forEach(function(matchedElement) {
+        var newAstDeclarations = matchedElement.newAstDeclarations;
+        delete matchedElement.newAstDeclarations; // clean up XML AST
+
+        var newInlineCss = declarationsAstToInlineCss(newAstDeclarations);
+        // Note: We don't minifiy or deduplicate anything here,
+        // this is not the scope of this plugin.
+        // Use minifyStyles plugin after this one.
+        matchedElement.attr('style').value = newInlineCss;
+
+        return;
+      });
+
+      return;
+    }
+
+    // Is this CSS AST rule without any selectors (empty selectors)?
+    function isRuleWithoutSelectors(astRule) {
+      return !(astRule.selectors && astRule.selectors.length > 0);
+    }
+
+    // Apply the stripped down styles on the styles elements
+    // Global variables used: styleElements
+    function reapplyStyleStyles() {
+      styleElements.forEach(function(styleElement) {
+
+        var styleElementAst = styleElement.stylesAst;
+        delete styleElement.stylesAst; // clean up XML AST (2)
+
+        cleanupStyleRules(styleElementAst);
+
+        var newStyleCss = stringifyAst(styleElementAst);
+        styleElement.content[0].text = newStyleCss;
+      });
+      return;
+    }
+
+    // Clean up style rules
+    function cleanupStyleRules(styleElementAst) {
+      // clean up rules that were stripped down to no selectors at all
+      styleElementAst.stylesheet.rules.forEach(function(styleAstRule) {
+        if(!isTopRule(styleAstRule)) {
+          return; // skip
+        }
+
+        if(isRuleWithoutSelectors(styleAstRule)) {
+          removeValue(styleElementAst.stylesheet.rules, styleAstRule);
+        }
+        return;
+      });
+    }
+
+
+    // Recursion
+
+    // Default: Always select an item
+    function selectAll() {
+      return true;
+    }
+    // Default: Always further recurse an item
+    function alwaysRecurse() {
+      return true;
+    }
+
+    /**
+     * Recursively traverses the HTML AST
+     *
+     * @param {Array}    items input items
+     * @param {Function} fn for processing an item
+     * @param {Function} fn for selecting an item
+     * @param {Function} fn for recursion check
+     *
+     * @return {Array} output items
+     */
+    function monkeys(items, callFn, selectFn, recurseFn) {
+      if(selectFn  === undefined) { selectFn  = selectAll;     }
+      if(recurseFn === undefined) { recurseFn = alwaysRecurse; }
+
+        var      i = 0,
+            length = items.content.length;
+
+        while(i < length) {
+
+            var item = items.content[i];
+
+            if ( selectFn(item) ) {
+              callFn(item);
+            }
+
+            // recurse
+            if(item.content && item.content.length > 0 &&

```

Comment:
You can use 'isEmpty()' method of API. See [lib/svgo/jsAPI.js#L87](https://github.com/svg/svgo/blob/master/lib/svgo/jsAPI.js#L87).